examples/parallel_memmap.py

```diff
@@ -1,19 +1,95 @@
 """
-==========================================
-Usage of numpy.memmap with joblib.Parallel
-==========================================
+===============================
+NumPy memmap in joblib.Parallel
+===============================
 
-This example shows how to preallocate data in memmap arrays both for input and
-output of the parallel worker processes.
+This example illustrates some features enabled by using :class:`numpy.memmap`
+within :class:`joblib.Parallel`. First, we show that dumping a huge data array
+ahead of passing it to :class:`joblib.Parallel` speed-up computation. Then, we
+show the possibility to provide write access to original data.
 
 """
-import tempfile
-import shutil
-import os
+
+##############################################################################
+# Speed-up processing with large data array
+##############################################################################
+# 
+# We create a large data array from which an average is computed for several
+# slices.
+
 import numpy as np
 
+data = np.random.random((int(1e7),))
+window_size = int(5e5)
+slices = [slice(start, start + window_size)
+          for start in range(0, data.size - window_size, int(1e5))]
+
+###############################################################################
+# The ``slow_mean`` function introduce a :func:`time.sleep` call to simulate a
+# more expensive computation cost for which parallel computing is beneficial.
+# Otherwise, for very fast sequential processing, parallel processing is not
+# adapted due some extra overheads (workers creations, communication, etc.).
```

Comment:
Rather: (no "Otherwise") "Parallel may not be beneficial for very fast operation, due to extra overhead (worker creation, communication...)"